protocol/contracts/profit/src/contract.rs

```diff
@@ -51,15 +48,26 @@ pub fn instantiate(
     )
     .grant_to(&msg.timealarms)?;
 
-    State::new(Config::new(
-        msg.cadence_hours,
-        msg.treasury,
-        OracleRef::try_from(msg.oracle, deps.querier)?,
-        TimeAlarmsRef::new(msg.timealarms, deps.querier)?,
-    ))
-    .store(deps.storage)?;
+    let init_state = State::new_state(
+        Config::new(
+            msg.cadence_hours,
+            msg.treasury,
+            OracleRef::try_from(msg.oracle, deps.querier)?,
+            TimeAlarmsRef::new(msg.timealarms, deps.querier)?,
+        ),
+        msg.dex,
+    );
+
+    let state_response = StateMachineResponse {
+        response: MessageResponse::messages_only(init_state.enter()),
+        next_state: init_state,
+    };
```

Comment:
wondering what is the purpose of creating it if immediately after that decompose it into parts?